src/components/user/services/userProfileSrv.js

```diff
@@ -1,76 +1,94 @@
+                    }
+                    if (!profile.nickname) {
+                        profile.nickname = nickNameFromAuth;
+                    }
+                    if (!profile.createdTime) {
+                        profile.createdTime = Firebase.ServerValue.TIMESTAMP;
+                    }
 
-                        return globalStorage.set(profilePath, profile);
-                    });
-                });
+                    _this.setProfile(profile);
+                    return profile;
+                }
             });
         };
 
         this.getProfileByUserId = function (userId) {
             var userProfilePath = 'users/' + userId + '/profile';
-            return InfraConfigSrv.getGlobalStorage().then(function(globalStorage) {
-                return globalStorage.get(userProfilePath);
+            return refAuthDB.child(userProfilePath).once('value').then(function (snapshot) {
+                return snapshot.val();
             });
         };
 
         this.setProfile = function (newProfile) {
-            return InfraConfigSrv.getGlobalStorage().then(function(globalStorage) {
-                return globalStorage.set(profilePath, newProfile);
+            var authData = rootRef.getAuth();
+            var profilePath = 'users/' + authData.uid + '/profile';
+            return refAuthDB.child(profilePath).once('value').then(function (snapshot) {
+                var profile = snapshot.val();
+                return profile ? refAuthDB.child(profilePath).update(newProfile) : refAuthDB.child(profilePath).set(newProfile);
             });
+
         };
 
         this.getCurrUserId = function(){
-            return InfraConfigSrv.getGlobalStorage().then(function(GlobalStorage){
-                var ref = GlobalStorage.adapter.getRef('');
-                var authData = ref.getAuth();
-                return authData && authData.uid;
-            });
+            var authData = rootRef.getAuth();
+            return $q.when(authData.uid);
         };
 
         this.updateUserTeachWorksId = function(uid, userTeachWorksId){
-            return InfraConfigSrv.getGlobalStorage().then(function(GlobalStorage){
-                var path = 'users/' + uid + '/teachworksId';
-                return GlobalStorage.update(path, userTeachWorksId);
+            var path = 'users/' + uid + '/teachworksId';
+            return refAuthDB.child(path).once('value').then(function (snapshot) {
+                var teachWorksId = snapshot.val();
+                return teachWorksId ? refAuthDB.child(path).update(userTeachWorksId) : refAuthDB.child(path).set(userTeachWorksId);
             });
         };
 
         this.getUserTeachWorksId = function(uid){
-            return InfraConfigSrv.getGlobalStorage().then(function(GlobalStorage){
-                var path = 'users/' + uid + '/teachworksId';
-                return GlobalStorage.get(path);
+            var path = 'users/' + uid + '/teachworksId';
```

Comment:
look in infra.config.js, you need to define new authStorage that will use ENV.fbDataEndPoint